app/src/androidTest/java/com/karumi/katasuperheroes/SuperHeroeDetailActivityTest.java

```diff
@@ -0,0 +1,102 @@
+package com.karumi.katasuperheroes;
+
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.suitebuilder.annotation.LargeTest;
+
+import com.karumi.katasuperheroes.di.MainComponent;
+import com.karumi.katasuperheroes.di.MainModule;
+import com.karumi.katasuperheroes.model.SuperHero;
+import com.karumi.katasuperheroes.model.SuperHeroesRepository;
+import com.karumi.katasuperheroes.ui.view.SuperHeroDetailActivity;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+import it.cosenonjaviste.daggermock.DaggerMockRule;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.scrollTo;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static com.karumi.katasuperheroes.matchers.ToolbarMatcher.onToolbarWithTitle;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by jesus on 30/1/16.
+ */
+@RunWith(AndroidJUnit4.class) @LargeTest public class SuperHeroeDetailActivityTest {
+
+    @Rule
+    public DaggerMockRule<MainComponent> daggerRule =
+            new DaggerMockRule<>(MainComponent.class, new MainModule()).set(
+                    new DaggerMockRule.ComponentSetter<MainComponent>() {
+                        @Override
+                        public void setComponent(MainComponent component) {
+                            SuperHeroesApplication app =
+                                    (SuperHeroesApplication) InstrumentationRegistry.getInstrumentation()
+                                            .getTargetContext()
+                                            .getApplicationContext();
+                            app.setComponent(component);
+                        }
+                    });
+
+    @Rule
+    public ActivityTestRule<SuperHeroDetailActivity> activityRule =
+            new ActivityTestRule<>(SuperHeroDetailActivity.class, true, false);
+
+    @Mock
+    SuperHeroesRepository repository;
+
+
+    @Test public void showsSuperHeroeNameAsToolbarTitle(){
+
+
+        SuperHero machote = givenThereIsASuperHero();
+        startActivity(machote);
+
+        onToolbarWithTitle(machote.getName()).check(matches(isDisplayed()));
+
+    }
+
```

Comment:
I miss some tests checking the information of this super hero is shown as expected, like the name and the description or the avengers badge.